### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by amit-s19**

### CODE REVIEW

File: admin/resources/school/SchoolCreate.tsx [hunk 1]
```diff
@@ -40,6 +48,62 @@ const SchoolCreate = () => {
...
+            type: data.type ? data.type : null,
+            udise: data.udise ? data.udise : null,
+          }
+        })
+        schoolRes = await schoolRes.json();
+        if (schoolRes?.data?.data?.returning?.[0]) {
+          notify(`School created successfully`, { type: 'success' });
+          redirect(`/school`);
+        } else if (schoolRes.errors) {
+          notify(`${schoolRes.errors[0].message}`, { type: 'error' });
```
Comment (geeky-abhishek) [inline admin/resources/school/SchoolCreate.tsx:99]: Always use optional chaining for such condition like schoolRes?.errors?.[0]?.message

File: admin/resources/school/SchoolCreate.tsx [hunk 2]
```diff
@@ -40,6 +48,62 @@ const SchoolCreate = () => {
...
+        }
+      }
+    } else {
+      notify(`Unable to create school: ${err.toString()}`, { type: 'error' });
```
Comment (geeky-abhishek) [inline admin/resources/school/SchoolCreate.tsx:103]: err.toString() or err.message ?

File: admin/resources/school/SchoolCreate.tsx [hunk 3]
```diff
@@ -103,28 +167,32 @@ const SchoolCreate = () => {
     }).sort((a: any, b: any) => { return a.name < b.name ? -1 : 1 });;
   }, [selectedBlock, districtData]);
```
Comment (geeky-abhishek) [inline admin/resources/school/SchoolCreate.tsx:169]: remove the log before pushing the code

File: admin/resources/school/SchoolCreate.tsx [hunk 4]
```diff
@@ -103,28 +167,32 @@ const SchoolCreate = () => {
...
 
   return (
-    <Create mutationOptions={{ onSuccess }}>
+    <Create mutationOptions={{ onSuccess, onError }}>
       <SimpleForm>
-        <TextInput source="name" validate={inputConstraints.fullName} />
-        <NumberInput source="udise" validate={required("Please enter a valid UDISE")} />
-        {/* <SelectInput label="District" source="location.data.district" onChange={(e: any) => {
+        <TextInput source="name" validate={inputConstraints.fullName} onChange={e => setData((prevData: any) => { return { ...prevData, name: e.target.value } })} />
+        <NumberInput source="udise" validate={required("Please enter a valid UDISE")} onChange={e => setData((prevData: any) => { return { ...prevData, udise: e.target.value } })} />
```
Comment (geeky-abhishek) [inline admin/resources/school/SchoolCreate.tsx:177]: it's correct but can we written in more clear way like 
`onChange={e => setData((prevData: ProperType) => ({ ...prevData, name: e.target.value })} `

File: admin/resources/school/SchoolCreate.tsx (hunk 2 above)
Comment (amit-s19) [reply admin/resources/school/SchoolCreate.tsx:103]: err.toString() only, err.message doesn't work here.

File: admin/resources/school/SchoolCreate.tsx (hunk 4 above)
Comment (amit-s19) [reply admin/resources/school/SchoolCreate.tsx:177]: okay

**Pull request merged**